averbis/core/_rest_client.py

```diff
@@ -811,27 +837,106 @@ def export_text_analysis(
         type_system = load_typesystem(
             self.project.client._export_analysis_result_typesystem(
-                self.project.name, self.document_source_name, document_id, self
+                self.project.name, self.document_source_name,
+                document_identifier, self
             )
         )
 
         # noinspection PyProtectedMember
         return load_cas_from_xmi(
             self.project.client._export_analysis_result_to_xmi(
-                self.project.name, self.document_source_name, document_id, self
+                self.project.name, self.document_source_name, document_name, self
             ),
             typesystem=type_system,
         )
 
+    @experimental_api
+    def add_text_analysis_result_to_document(
+            self,
+            source: Union[Cas, Path, IO],
+            document_name: str,
+            mime_type: str = None,
+            typesystem: "TypeSystem" = None
+    ):
+        """
+        HIGHLY EXPERIMENTAL API - may soon change or disappear.
+
+        Add a text analysis result (i.e. annotated content) to a specified document in this process.
+        This process must be a manual or imported process (i.e. without automatic processing via pipeline).
+        There must not already exist a text analysis result associated
+        with this document, use Process.update_text_analysis_for_document() instead.
+
+        The supported file content types (mime_types) are UIMA CAS XMI (application/vnd.uima.cas+xmi),
+        XCAS (application/vnd.uima.cas+xcas), binary CAS (application/vnd.uima.cas+binary),
+        binary TSI (application/vnd.uima.cas+binary.tsi), compressed (application/vnd.uima.cas+compressed),
+        compressed TSI (application/vnd.uima.cas+compressed.tsi),
+        compressed filtered (application/vnd.uima.cas+compressed.filtered),
+        compressed filtered TS (application/vnd.uima.cas+compressed.filtered.ts),
+        compressed filtered TSI (application/vnd.uima.cas+compressed.filtered.tsi),
+        serialized CAS (application/vnd.uima.cas+serialized)
+        and serialized TSI (application/vnd.uima.cas+serialized.tsi).
+
+        If a document is provided as a CAS object, the type system information can be automatically picked from the CAS
+        object and should not be provided explicitly. The mime_type is also not needed.
+        If a CAS is provided as a path or stream, then a mime_type needs to be given. The typesystem might need to be
+        provided depending on the content type.
+
+        """
+        # noinspection PyProtectedMember
+        self.project.client._add_text_analysis_result_to_document(self.project.name, self.name,
+                                                                  self.document_source_name, source, document_name,
+                                                                  mime_type, typesystem)
+
+    @experimental_api
+    def update_text_analysis_result_for_document(
```

Comment:
The update is actually an "upsert", right? To stick with existing naming in the Python client, let's call it `upload_text_analysis_result` or `import_text_analysis_results` and use a parameter `overwrite` (default to false) to control whether it calls the "add only" or the "upsert" endpoint.